src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c

```diff
@@ -21,69 +87,1182 @@ const ActorInit Boss_Fd2_InitVars = {
+        *dList = D_06004E38;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/func_808D32B4.s")
+    if ((limbIndex == 32) || (limbIndex == 35) || (limbIndex == 36)) {
+        OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_fd2.c", 0x875);
+        gDPPipeSync(oGfxCtx->polyOpa.p++);
+        gDPSetEnvColor(oGfxCtx->polyOpa.p++, 0xFF, 0xFF, 0xFF, (s8)bossFd->headTex2Opacity);
+        CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_fd2.c", 0x87C);
+    } else {
+        OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_fd2.c", 0x87E);
+        gDPPipeSync(oGfxCtx->polyOpa.p++);
+        gDPSetEnvColor(oGfxCtx->polyOpa.p++, 0xFF, 0xFF, 0xFF, (s8)bossFd->bodyTex2Opacity);
+        CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_fd2.c", 0x885);
+    }
+    if ((limbIndex > 0) && (limbIndex < 16)) {
+        *dList = NULL;
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/func_808D3330.s")
+static Vec3f D_808D620C = { 4500.0f, 0.0f, 0.0f };
+static Vec3f D_808D6218 = { 4000.0f, 0.0f, 0.0f };
+static Vec3f D_808D6224 = { 4000.0f, -2900.0, 2000.0f };
+static Vec3f D_808D6230 = { 4000.0f, -1600.0, 0.0f };
+static Vec3f D_808D623C = { 4000.0f, -1600.0, -2000.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/func_808D3400.s")
+void BossFd2_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    BossFd2* this = THIS;
+    
+    if (limbIndex == 35) {
+        Matrix_MultVec3f(&D_808D620C, &this->actor.posRot2.pos);
+        Matrix_MultVec3f(&D_808D6218, &this->fireBreathSpawnPoint);
+        Matrix_MultVec3f(&D_808D6224, &this->centerMane.head);
+        Matrix_MultVec3f(&D_808D6230, &this->rightMane.head);
+        Matrix_MultVec3f(&D_808D623C, &this->leftMane.head);
+    }
+    func_800628A4(limbIndex, &this->collider1);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/func_808D3464.s")
+void BossFd2_UpdateMane(BossFd2* this, GlobalContext* globalCtx, Vec3f* head, Vec3f* pos, Vec3f* rot, Vec3f* pull,
+                        f32* scale) {
+    f32 sp138[] = { 0.0f, 100.0f, 50.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f };
+    f32 sp110[] = { 0.0f, 5.0f, -10.0f, 500.0f, 500.0f, 500.0f, 500.0f, 500.0f, 500.0f, 500.0f };
+    f32 spE8[] = { 0.4f, 0.6f, 0.8f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f };
+    f32 temp_f12;
+    f32 temp_f23;
+    f32 temp_f5;
+    f32 temp_f20;
+    f32 temp_f24;
+    f32 temp_f2;
+    f32 phi_f0;
+    f32 temp_f22;
+    Vec3f spBC;
+    Vec3f spB0;
+    s16 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/func_808D39A8.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_fd2.c", 0x955);
+    Matrix_Push();
+    gDPPipeSync(oGfxCtx->polyOpa.p++);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd2/func_808D3A08.s")
+    for (i = 0; i < 10; i++) {
+        if (i == 0) {
+            (pos + i)->x = head->x;
```

Comment:
`pos[i].x` doesn't match? I would think it's more natural but idk

Reply:
it doesn't. weird, but true.